tests/ttsim/interface_dag_elements/test_failures.py

```diff
@@ -930,6 +930,19 @@ def c(b):
         )
 
 
+def test_fail_if_input_data_is_missing_via_main(backend):
+    with pytest.raises(
+        ValueError,
+        match=r"\('input_data', 'flat'\)",
+    ):
+        main(
+            main_target=MainTarget.specialized_environment.tax_transfer_dag,
```

Comment:
```suggestion
            main_target=MainTarget.specialized_environment.tax_transfer_function,
```

Just to be sure, please add another test where we try to go end-to-end without input data. That is the most common case, I guess.